woocommerce/pinterest-for-woocommerce
# Feature/merchant health check

Closes #160 
Closes #141
Closes #51 

### Changes proposed in this pull request

The idea is to improve user facing error messages and present to the user the current state of the Merchant object of the Pinterest account, since that's where the context of an approved/pending/declined account lives. 

Adds a HealthCheck component to the UI which asks from the backend the status of the Merchant and any "reasons" for a possible account disapproval. This renders a notice which should so a clear picture to the user of what state is the account in. 
Also in the case of a disapproved account, the user can click a link to submit an appeal.

Also during the "Claim your website" step of the onboarding guide, we are now printing right below the domain, the error message returned from the API, provided the error code matches some [preset codes.](https://github.com/woocommerce/pinterest-for-woocommerce/pull/227/files#diff-2eb4c0f8c84efca988abbd09715322310b9fd3ae01958dfb86c767957f218249R32) 

#### Screenshots


**Print out the error message during domain verification.**
![Screenshot 2021-10-18 231107](https://user-images.githubusercontent.com/4016167/137800688-1e9e3f87-ae0b-4bbe-947b-8c0e4bd25013.png)

---

**As we cannot configure and get status of the merchant, without registering the product feed (current limitation of the Pinterest API), and we cannot register the product feed until it is fully generated, there is a wait period till we get an actual status:**
![Screenshot 2021-10-19 170926](https://user-images.githubusercontent.com/4016167/137927283-61988a5a-93b8-4311-97d5-79552179f1c4.png)

---

**Error message on a declined account:** 
![Screenshot 2021-10-18 231355](https://user-images.githubusercontent.com/4016167/137800684-d675c68a-04aa-4de1-9f3d-dc45b72ac27f.png)

---

**Appeal pending message**
![Screenshot 2021-10-18 231627](https://user-images.githubusercontent.com/4016167/137800690-6158405d-76a4-4f5d-b6db-f76f1604a41a.png)

---

**Declined message with reasons**
![Screenshot 2021-10-18 231512](https://user-images.githubusercontent.com/4016167/137800691-fe6f680e-47fb-447a-b484-fa531e6cf12b.png)

### Detailed test instructions

1. Start with a clean plugin installation and a new Pinterest account.
2. Start the onboarding process and Connect to Pinterest.
3. During the "Claim your website" step, attempt verification of an already claimed by another user domain and verify the error message.
4. Complete onboarding process
5. Verify that when you are navigated to the plugin's settings after completing the onboarding, you are presented with the status of the account (Which should be "pending" on a new account).
6. Verify that after your account gets disapproved (New accounts usually do), you can see that in the HealthCheck notice and you can submit an appeal by clicking the link in the notice. 
7. Verify that when your account is approved, there is notice shown to the user.
